### PR TITLE
[enhancement](alter) Make alter job more robust by ignoring some task failure

### LINKED ISSUE #10718: [Enhancement] (alter) Increase the success rate of alter job by ignoring some task failure
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/incubator-doris/issues?q=is%3Aissue) and found no similar issues.

### Description

now, when we do some alter operation on big table, it is easy to see some task failure, but we shouldn't 
simply mark the job as failed, we allow some task failure because of multiple replicas for one tablet.

### Solution

change the logic of marking failure of alter job

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by caiconghui**
# Proposed changes

Issue Number: close #10718 

## Problem Summary:

Describe the overview of changes.

## Checklist(Required)

1. Does it affect the original behavior: (Yes/No/I Don't know)
2. Has unit tests been added: (Yes/No/No Need)
3. Has document been added or modified: (Yes/No/No Need)
4. Does it need to update dependencies: (Yes/No)
5. Are there any changes that cannot be rolled back: (Yes/No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/alter/RollupJobV2.java [hunk 1]
```diff
@@ -444,6 +456,12 @@ protected void runRunningJob() throws AlterCancelException {
         tbl.writeLockOrAlterCancelException();
         try {
             Preconditions.checkState(tbl.getState() == OlapTableState.ROLLUP);
+            TabletInvertedIndex invertedIndex = Catalog.getCurrentInvertedIndex();
+            for (List<AgentTask> tasks : failedAgentTasks.values()) {
+                for (AgentTask task : tasks) {
+                    invertedIndex.getReplica(task.getTabletId(), task.getBackendId()).setBad(true);
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/alter/RollupJobV2.java:462]: This tablet id is new rollup tabeltid  or base tablet id?

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/alter/RollupJobV2.java:462]: new rollup tablet id

File: fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java [hunk 2]
```diff
@@ -494,22 +492,38 @@ protected void runRunningJob() throws AlterCancelException {
...
+                    AgentTaskQueue.removeTask(task.getBackendId(), TTaskType.ALTER, task.getSignature());
+                    LOG.warn("schema change task failed after try three times: " + task.getErrorMsg());
+                    if (!failedAgentTasks.containsKey(task.getTabletId())) {
+                        failedAgentTasks.put(task.getTabletId(), Lists.newArrayList(task));
+                    } else {
+                        failedAgentTasks.get(task.getTabletId()).add(task);
+                    }
+                    int expectSucceedTaskNum = tbl.getPartitionInfo()
+                            .getReplicaAllocation(task.getPartitionId()).getTotalReplicaNum();
+                    int failedTaskCount = failedAgentTasks.get(task.getTabletId()).size();
```
Comment (yiguolei) [inline fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:505]: why not use the replica num in config?

Comment (caiconghui) [reply fe/fe-core/src/main/java/org/apache/doris/alter/SchemaChangeJobV2.java:505]: we are not sure the status of all agent tasks, but we can confirm that replica with failed task is bad, the job is still in running state

**Review by yiguolei — APPROVED** [review]
LGTM

**Review by morningman — APPROVED** [review]
LGTM
But did you test it in real env? How to test it?

**Comment by caiconghui** [comment]
> LGTM But did you test it in real env? How to test it?



> LGTM But did you test it in real env? How to test it?

not yet, but it can simply stop one be after doing some alter operation, then what we expected is final successful alter job

**Pull request merged**